test/javascript/components/profile/contribution-summary/track-switcher.test.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react'
+
+import { render, fireEvent, act, screen, within } from '@testing-library/react'
+import '@testing-library/jest-dom/extend-expect'
+
+import { TrackSwitcher } from '../../../../../app/javascript/components/profile/contributions-summary/TrackSwitcher'
+
+import {
+  createTrack,
+  totalTrackReputation,
+} from '../../../factories/ContributionSummaryFactory'
+
+const setup = () => {
+  const tracks = [
+    createTrack(),
+    createTrack({
+      id: 'elixir',
+      title: 'Elixir',
+      iconUrl: '/icons/elixir.svg',
+    }),
+  ]
+
+  return {
+    tracks,
+  }
+}
+
+describe('TrackSwitcher', () => {
+  test('shows value on render', () => {
+    const { tracks } = setup()
+
+    const currentTrack = tracks[0]
+
+    render(
+      <TrackSwitcher tracks={tracks} value={currentTrack} setValue={() => {}} />
+    )
+
+    screen.getByAltText(`icon for ${currentTrack.title} track`)
+    screen.getByText(currentTrack.title)
+    screen.getByText(`${totalTrackReputation(currentTrack)} rep`)
+  })
+
+  test('shows dropdown on click', async () => {
+    const { tracks } = setup()
+
+    const currentTrack = tracks[0]
+
+    render(
+      <TrackSwitcher tracks={tracks} value={currentTrack} setValue={() => {}} />
+    )
+
+    const button = screen.getByLabelText('Open the track switcher')
+
+    await act(async () => {
+      await fireEvent.click(button)
```

Comment:
I usually use `userEvent.click` instead of `fireEvent.click`. Is there any advantage to use `fireEvent` instead?

Reply:
I use fireEvent because that's what I know, but looking for a reference to think points out that using userEvent might be a better way to go about this. Reference: https://github.com/testing-library/eslint-plugin-testing-library/issues/162#issuecomment-645135887

Reply:
@neenjaw Want to change or should I merge? :)

Reply:
They are essentially the same thing, so this works and covers it, but if I'm in this piece of code again I would change it. I think good to merge